### PR TITLE
test(ci): adding wrapper for vg and pv operations

### PR DESCRIPTION
**Pull request by abhilashshetty04**
**Why is this PR required? What issue does it fix?**:
This PR adds required wrapper funcations for PV and VG operations. Its required for new tests to be automated

**What this PR does?**:
Same as above

**Does this PR require any upgrade changes?**:
NA

**Checklist:**
- [ ] Fixes #<issue number>
- [ ] PR Title follows the convention of  `<type>(<scope>): <subject>`
- [ ] Has the change log section been updated?
- [x] Commit has unit tests
- [ ] Commit has integration tests
- [ ] (Optional) Are upgrade changes included in this PR? If not, mention the issue/PR to track:
- [ ] (Optional) If documentation changes are required, which issue on https://github.com/openebs/openebs-docs is used to track them:

### CODE REVIEW

**Comment by dsharma-dc** [comment]
What is the rationale behind these wrappers? Sometime earlier we discussed that we'll leave these operations as it exists today, until we integrate to a common mechanism to manage lvm including Mayastor LVM backend.

**Comment by abhilashshetty04** [comment]
> What is the rationale behind these wrappers? Sometime earlier we discussed that we'll leave these operations as it exists today, until we integrate to a common mechanism to manage lvm including Mayastor LVM backend.

This is going to be used in the ci tests. We have scenarios where there is multiple vg in the node, there is a scenario to do vgextend ans see provisioning work.

File: tests/utils.go [hunk 1]
```diff
@@ -736,6 +736,98 @@ func enableThinpoolMonitoring() {
...
+
+	args_loop := []string{
+		"losetup", "-d",
+		device,
+	}
+	_, _, err_loop := execAtLocal("sudo", nil, args_loop...)
+	gomega.Expect(err_loop).To(gomega.BeNil(), "loop device remove failed")
+	filename := fmt.Sprintf("/tmp/openebs_ci_disk_%s.img", file_id)
+	args_file := []string{
+		"rm",
```
Comment (dsharma-dc) [inline tests/utils.go]: you may want to do `rm -f` here

File: tests/utils.go [hunk 2]
```diff
@@ -736,6 +736,98 @@ func enableThinpoolMonitoring() {
 	gomega.Expect(err).To(gomega.BeNil(), "run lvchange command")
 }
 
+// This creates loopdevice using file_id(for unquiness) and size,
+// Uses the new loop device to create PV. returns loopdevice name to the caller.
+func createPV(size int, file_id string) string {
```
Comment (dsharma-dc) [inline tests/utils.go]: I suggest making it more generic like - don't pass any `file_id` string. Assign a random string(using a generator) to concatenate on file name e.g. `openebs_ci_disk_fXVftN.img`
While deleting in `removePV` only pass the loop device name and get the back-file for it to delete the file by filtering using `-O` flag .

File: tests/utils.go (hunk 1 above)
Comment (abhilashshetty04) [reply tests/utils.go]: Done

File: tests/utils.go (hunk 2 above)
Comment (abhilashshetty04) [reply tests/utils.go]: Done. Using timestamp for uniqueness

**Review by dsharma-dc — APPROVED** [review]

File: tests/utils.go [hunk 3]
```diff
@@ -776,6 +776,109 @@ func enableThinpoolMonitoring() {
...
+		size_str, filename,
+	}
+	_, _, err := execAtLocal("sudo", nil, args_file...)
+	gomega.Expect(err).ShouldNot(gomega.HaveOccurred(), "create device failed")
+	args_loop := []string{
+		"losetup", "-f",
+		filename, "--show",
+	}
+
+	stdout_loop, _, err := execAtLocal("sudo", nil, args_loop...)
```
Comment (niladrih) [inline tests/utils.go]: We can run the test using sudo, e.g.: https://github.com/openebs/dynamic-localpv-provisioner/blob/v3.5.x/Makefile#L146

If the test is using a shell script, e.g. test.sh, run it using `sudo bash -c "./test.sh"`

We should not call sudo implicitly like so.

File: tests/utils.go
Comment (niladrih) [inline tests/utils.go:1]: nit: you could put these lvm operations in a separate file called lvm_ops.go or lvm.go or disk.go or something like that.

**Comment by sinhaashish** [comment]
can you please squash the relevant commits. I see commits `856c95e83e12a4bdd2cb8d6a5e6b2a1e7ce80db3`  and `f1a328cd2758e5a8f52ca0aabde3abc46a91d73b` are modifying the same file.

File: tests/provision_test.go [hunk 4]
```diff
@@ -190,12 +190,26 @@ func leakProtectionTest() {
 	By("Deleting storage class", deleteStorageClass)
 }
 
+// This acts as a test just to call the new wrapper functions,
+// Once we write tests calling them this will not be required.
+// This doesnt test any Openebs component.
+func lvmOps() {
```
Comment (niladrih) [inline tests/provision_test.go:196]: Why is this required? Isn't there a script which sets these up already?

Comment (abhilashshetty04) [reply tests/provision_test.go:196]: No that will be in next PR.

File: tests/utils.go (hunk 2 above)
Comment (niladrih) [reply tests/utils.go]: Maybe use `mktemp` or Go's Tempfile API, e.g. https://github.com/openebs/dynamic-localpv-provisioner/blob/v3.5.x/tests/suite_test.go#L95?

File: tests/utils.go (hunk 3 above)
Comment (dsharma-dc) [reply tests/utils.go]: The main ci script in lvm repo runs as non-root user in github actions. Most of the work done inside that script is doing `sudo` today e.g. `sudo vgcreate`. These new wrappers seem to be emulating same.

File: tests/utils.go
Comment (abhilashshetty04) [reply tests/utils.go:1]: Done, moved to lvm_util.go

File: tests/utils.go (hunk 3 above)
Comment (niladrih) [reply tests/utils.go]: This feels like tech debt. But oh well.

Comment (tiagolobocastro) [reply tests/utils.go]: We could have a wrapper for calls which require sudo. In general IMHO it's better to use sudo only for operations which need it rather than blindly run everything with sudo as that seems a lot more dangerous.
